homeassistant/components/blue_current/config_flow.py

```diff
@@ -0,0 +1,62 @@
+"""Config flow for Blue Current integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from bluecurrent_api import Client
+from bluecurrent_api.exceptions import (
+    AlreadyConnected,
+    InvalidApiToken,
+    RequestLimitReached,
+    WebsocketError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN, LOGGER
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_API_TOKEN): str})
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the config flow for Blue Current."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            client = Client()
+            api_token = user_input[CONF_API_TOKEN]
+            self._async_abort_entries_match(user_input)
+
```

Comment:
```suggestion
```

Even tho it's a valid reason to stop here if the api key is already present, we are also checking with the customer_id later on, so I don't think this really think this is needed here at the moment